### PR TITLE
Cmake fixes

### PR DESCRIPTION
**Pull request by MathiasMagnus**
This PR fixes a typo in build interface include dirs and deduplicates some login in test and benchmark definitions.

### CODE REVIEW

**Comment by AlexVlx** [comment]
@MathiasMagnus  thank you for submitting this, but I believe that streams might've gotten crossed? For example, we don't want to disable Linux & Windows CI testing - perhaps this needs a bit of a cleanup / separation?

**Comment by MathiasMagnus** [comment]
Yes, apologies, this is still a WIP. CI is only enabled for the master branch and not for working branches. Could we borrow the [Draft PR feature](https://github.blog/2019-02-14-introducing-draft-pull-requests/) of GitHub? I don't think it's enabled for this repo. It works very well on Vcpkg and is useful for messaging that something is half baked.

As a contributor, when I'm opening a PR, there are two buttons, one is regular PR and the other is Draft PR. It's similar to starting the PR name with "WIP:" on GitLab. The UI changes and clearly states it's not meant for merging, but allows feedback from upstream whether they like the design or not.

**Comment by MathiasMagnus** [comment]
I Linux and Windows and many things to speed up CI runtimes while all tests are supposed to fail building/executing the same way. I intended to turn them back on once the work is done.

**Comment by AlexVlx** [comment]
> Yes, apologies, this is still a WIP. CI is only enabled for the master branch and not for working branches. Could we borrow the [Draft PR feature](https://github.blog/2019-02-14-introducing-draft-pull-requests/) of GitHub? I don't think it's enabled for this repo. It works very well on Vcpkg and is useful for messaging that something is half baked.
> 
> As a contributor, when I'm opening a PR, there are two buttons, one is regular PR and the other is Draft PR. It's similar to starting the PR name with "WIP:" on GitLab. The UI changes and clearly states it's not meant for merging, but allows feedback from upstream whether they like the design or not.

I'm afraid that we're not going to have Draft PRs for the time being, since as far as I can gather they're not available for the sort of account that's underpinning this repository. Perhaps at some point in the future, it is a very useful feature.

**Comment by MathiasMagnus** [comment]
@AlexVlx Requesting review. I have removed some code duplication in test and example code definitions, also finished moving away from manual control of include and link library additions (especially those added through global state) and consolidated picking up dependencies to imported targets. Build flags are also managed through imported target with compiler-specific branches handles through generator expressions instead of script logic.

The only (intended) functional difference is adding `/W4` to the MSVC build flags, as other compilers tend to be used in a similar fashion. Let me know if that's too much. There are a few interesting messages on the console:
- `warning C5105: macro expansion producing 'defined' has undefined behavior`
- `warning C4068: unknown pragma 'unroll'`

### Segue

FWIW I would completely move away from adding flags via build scripts which aren't essential for the compilation to succeed. I didn't know what was missing from my (professional) life until I came across [this tweet](https://twitter.com/JSAMcFarlane/status/1252330666722758658?s=20). Build scripts (ideally, at least in my ideal world) would define build requirements only (required lang standards, ABI altering reqs, etc.), not build preferences (diagnostic, optimization levels). It may be something worth considering.

One hot topic internally was adding `-Werror` and `/WX` to the scripts by default, because we want to compile warning-free at given warning levels. It constantly tripped people over, who were using newer compilers than the ones others CI was using and it made everyone be constantly adding workarounds to warnings others introduced which CI did not alert us to, because CI images usually don't use bleeding edge compilers.

Long story short (IMHO):
- if libs should compile warning free, add `-Werror`, `/WX` and the target warning levels/flags to CI scripts _only_ and don't trip over casual devs cloning your repo.
- if devs want to see `info C5002: loop not vectorized due to reason '500'` messages, configure CI/devenv accordingly. (I guess this is produced by `/Qvec-report:2`.
- is `/arch:AVX2` a build requirement? Should a project wait for the first Windows user targeting ARM64 to face conflicting arch definitions or should these be in CI only where the execution environment is known?

File: examples/CMakeLists.txt [hunk 1]
```diff
@@ -1,12 +1,31 @@
...
+add_library(examples_common INTERFACE)
 
-if (${CMAKE_HOST_APPLE}) # Add brew-ed TBB.
-    include_directories(/usr/local/include)
-    link_directories(/usr/local/lib)
-endif()
+target_link_libraries(examples_common
+    INTERFACE
+        hip_cpu_rt::hip_cpu_rt
+)
```
Comment (AlexVlx) [inline examples/CMakeLists.txt:8]: Please re-format this to match the rest of the codebase i.e. no trailing parentheses, clear newline after the opening parenthese and no extra vertical spacing (reckon the full line might fit within the 80 character limit but, if not, the argument to `target_link_libraries` will fit once you've put it on its own line)

File: examples/CMakeLists.txt [hunk 2]
```diff
@@ -1,12 +1,31 @@
...
+target_compile_options(examples_common
```
Comment (AlexVlx) [inline examples/CMakeLists.txt:9]: I don't think we really need these, they're historical artifacts I didn't clean up before going public, these should be under user control and not enforced.

File: examples/CMakeLists.txt [hunk 3]
```diff
@@ -1,12 +1,31 @@
...
+            /arch:AVX2
+            /permissive-
+            /Qvec-report:2
+            /Zc:preprocessor
+        >
+        $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:
+            -Wall
+        >
+)
+target_compile_definitions(examples_common
```
Comment (AlexVlx) [inline examples/CMakeLists.txt:22]: We do need this, but please match style.

File: examples/bit_extract/CMakeLists.txt [hunk 4]
```diff
@@ -2,19 +2,6 @@ set(name bit_extract)
...
-    target_compile_definitions(${name} PRIVATE WIN32_LEAN_AND_MEAN NOMINMAX)
-else ()
-    target_compile_options(${name} PRIVATE -Wall -fopenmp-simd)
-    target_link_libraries(${name} PRIVATE dl tbb)
-endif ()
-target_include_directories(
-    ${name} PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/../../include)
-target_link_libraries(${name} PRIVATE Threads::Threads)
\ No newline at end of file
+target_link_libraries(${name}
```
Comment (AlexVlx) [inline examples/bit_extract/CMakeLists.txt:5]: Please match style.

File: examples/hip_bus_bandwidth/CMakeLists.txt [hunk 5]
```diff
@@ -2,19 +2,6 @@ set(name hip_bus_bandwidth)
...
-    target_compile_definitions(${name} PRIVATE WIN32_LEAN_AND_MEAN NOMINMAX)
-else ()
-    target_compile_options(${name} PRIVATE -fopenmp-simd -Wall)
-    target_link_libraries(${name} PRIVATE dl tbb)
-endif ()
-target_include_directories(
-    ${name} PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/../../include)
-target_link_libraries(${name} PRIVATE Threads::Threads)
\ No newline at end of file
+target_link_libraries(${name}
```
Comment (AlexVlx) [inline examples/hip_bus_bandwidth/CMakeLists.txt:5]: Please match style.

File: examples/hip_dispatch_enqueue_rate_mt/CMakeLists.txt [hunk 6]
```diff
@@ -2,19 +2,6 @@ set(name hip_dispatch_enqueue_rate_mt)
...
-    target_compile_definitions(${name} PRIVATE WIN32_LEAN_AND_MEAN NOMINMAX)
-else ()
-    target_compile_options(${name} PRIVATE -fopenmp-simd -Wall)
-    target_link_libraries(${name} PRIVATE dl tbb)
-endif ()
-target_include_directories(
-    ${name} PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/../../include)
-target_link_libraries(${name} PRIVATE Threads::Threads)
\ No newline at end of file
+target_link_libraries(${name}
```
Comment (AlexVlx) [inline examples/hip_dispatch_enqueue_rate_mt/CMakeLists.txt:5]: Please match style.

File: examples/hip_dispatch_latency/CMakeLists.txt [hunk 7]
```diff
@@ -2,19 +2,6 @@ set(name hip_dispatch_latency)
...
-    target_compile_definitions(${name} PRIVATE WIN32_LEAN_AND_MEAN NOMINMAX)
-else ()
-    target_compile_options(${name} PRIVATE -fopenmp-simd -Wall)
-    target_link_libraries(${name} PRIVATE dl tbb)
-endif ()
-target_include_directories(
-    ${name} PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/../../include)
-target_link_libraries(${name} PRIVATE Threads::Threads)
\ No newline at end of file
+target_link_libraries(${name}
```
Comment (AlexVlx) [inline examples/hip_dispatch_latency/CMakeLists.txt:5]: Please match style.

File: examples/hip_event/CMakeLists.txt [hunk 8]
```diff
@@ -2,20 +2,6 @@ set(name hip_event)
...
-    target_compile_definitions(${name} PRIVATE WIN32_LEAN_AND_MEAN NOMINMAX)
-else ()
-    target_compile_options(${name} PRIVATE -fopenmp-simd -Wall)
-    target_link_libraries(${name} PRIVATE dl tbb)
-endif ()
-target_include_directories(
-    ${name} PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/../../include)
-target_link_libraries(${name} PRIVATE Threads::Threads)
\ No newline at end of file
+target_link_libraries(${name}
```
Comment (AlexVlx) [inline examples/hip_event/CMakeLists.txt:5]: Please match style.

File: examples/hip_info/CMakeLists.txt [hunk 9]
```diff
@@ -2,20 +2,6 @@ set(name hip_info)
...
-    target_compile_definitions(${name} PRIVATE WIN32_LEAN_AND_MEAN NOMINMAX)
-else ()
-    target_compile_options(${name} PRIVATE -fopenmp-simd -Wall)
-    target_link_libraries(${name} PRIVATE dl tbb)
-endif ()
-target_include_directories(
-    ${name} PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/../../include)
-target_link_libraries(${name} PRIVATE Threads::Threads)
\ No newline at end of file
+target_link_libraries(${name}
```
Comment (AlexVlx) [inline examples/hip_info/CMakeLists.txt:5]: Please match style.

File: examples/matrix_transpose/CMakeLists.txt [hunk 10]
```diff
@@ -2,20 +2,6 @@ set(name matrix_transpose)
...
-    target_compile_definitions(${name} PRIVATE WIN32_LEAN_AND_MEAN NOMINMAX)
-else ()
-    target_compile_options(${name} PRIVATE -fopenmp-simd -Wall)
-    target_link_libraries(${name} PRIVATE dl tbb)
-endif ()
-target_include_directories(
-    ${name} PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/../../include)
-target_link_libraries(${name} PRIVATE Threads::Threads)
\ No newline at end of file
+target_link_libraries(${name}
```
Comment (AlexVlx) [inline examples/matrix_transpose/CMakeLists.txt:5]: Please match style.

File: examples/occupancy/CMakeLists.txt [hunk 11]
```diff
@@ -2,20 +2,6 @@ set(name occupancy)
...
-    target_compile_definitions(${name} PRIVATE WIN32_LEAN_AND_MEAN NOMINMAX)
-else ()
-    target_compile_options(${name} PRIVATE -fopenmp-simd -Wall)
-    target_link_libraries(${name} PRIVATE dl tbb)
-endif ()
-target_include_directories(
-    ${name} PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/../../include)
-target_link_libraries(${name} PRIVATE Threads::Threads)
\ No newline at end of file
+target_link_libraries(${name}
```
Comment (AlexVlx) [inline examples/occupancy/CMakeLists.txt:5]: Please match style.

File: examples/pragma_unroll/CMakeLists.txt [hunk 12]
```diff
@@ -2,20 +2,6 @@ set(name pragma_unroll)
...
-    target_compile_definitions(${name} PRIVATE WIN32_LEAN_AND_MEAN NOMINMAX)
-else ()
-    target_compile_options(${name} PRIVATE -fopenmp-simd -Wall)
-    target_link_libraries(${name} PRIVATE dl tbb)
-endif ()
-target_include_directories(
-    ${name} PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/../../include)
-target_link_libraries(${name} PRIVATE Threads::Threads)
\ No newline at end of file
+target_link_libraries(${name}
```
Comment (AlexVlx) [inline examples/pragma_unroll/CMakeLists.txt:5]: Please match style.

File: examples/shared_memory/CMakeLists.txt [hunk 13]
```diff
@@ -2,20 +2,6 @@ set(name shared_memory)
...
-    target_compile_definitions(${name} PRIVATE WIN32_LEAN_AND_MEAN NOMINMAX)
-else ()
-    target_compile_options(${name} PRIVATE -fopenmp-simd -Wall)
-    target_link_libraries(${name} PRIVATE dl tbb)
-endif ()
-target_include_directories(
-    ${name} PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/../../include)
-target_link_libraries(${name} PRIVATE Threads::Threads)
\ No newline at end of file
+target_link_libraries(${name}
```
Comment (AlexVlx) [inline examples/shared_memory/CMakeLists.txt:5]: Please match style.

File: examples/square/CMakeLists.txt [hunk 14]
```diff
@@ -2,20 +2,6 @@ set(name square)
...
-    target_compile_definitions(${name} PRIVATE WIN32_LEAN_AND_MEAN NOMINMAX)
-else ()
-    target_compile_options(${name} PRIVATE -fopenmp-simd -Wall)
-    target_link_libraries(${name} PRIVATE dl tbb)
-endif ()
-target_include_directories(
-    ${name} PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/../../include)
-target_link_libraries(${name} PRIVATE Threads::Threads)
\ No newline at end of file
+target_link_libraries(${name}
```
Comment (AlexVlx) [inline examples/square/CMakeLists.txt:5]: Please match style.

File: examples/streams/CMakeLists.txt [hunk 15]
```diff
@@ -2,20 +2,6 @@ set(name streams)
...
-    target_compile_definitions(${name} PRIVATE WIN32_LEAN_AND_MEAN NOMINMAX)
-else ()
-    target_compile_options(${name} PRIVATE -fopenmp-simd -Wall)
-    target_link_libraries(${name} PRIVATE dl tbb)
-endif ()
-target_include_directories(
-    ${name} PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/../../include)
-target_link_libraries(${name} PRIVATE Threads::Threads)
\ No newline at end of file
+target_link_libraries(${name}
```
Comment (AlexVlx) [inline examples/streams/CMakeLists.txt:5]: Please match style.

File: examples/vadd_hip/CMakeLists.txt [hunk 16]
```diff
@@ -2,20 +2,6 @@ set(name vadd_hip)
...
-    target_compile_definitions(${name} PRIVATE WIN32_LEAN_AND_MEAN NOMINMAX)
-else ()
-    target_compile_options(${name} PRIVATE -fopenmp-simd -Wall)
-    target_link_libraries(${name} PRIVATE dl tbb)
-endif ()
-target_include_directories(
-    ${name} PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/../../include)
-target_link_libraries(${name} PRIVATE Threads::Threads)
\ No newline at end of file
+target_link_libraries(${name}
```
Comment (AlexVlx) [inline examples/vadd_hip/CMakeLists.txt:5]: Please match style.

File: tests/CMakeLists.txt [hunk 17]
```diff
@@ -1,82 +1,97 @@
...
+    hip_host_memcpy_all.cpp
+    # hip_host_module.cpp
+    hip_host_multithread_hipstream_memcpy.cpp
+    hip_host_multithread_hipstream.cpp
+    hip_host_multithread_memcpy.cpp
+    hip_set_device_flags.cpp
+    hip_set_device.cpp
+    hip_vector_types.cpp
+)
+target_link_libraries(legacy_tests
```
Comment (AlexVlx) [inline tests/CMakeLists.txt:92]: Please match style.

File: tests/CMakeLists.txt [hunk 18]
```diff
@@ -1,82 +1,97 @@
...
+        $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:
+            -ffast-math
+        >
+)
+
+add_executable(performance_tests
+    catch_benchmarking_main.cpp
+    benchmarks.cpp
+)
+target_link_libraries(performance_tests
```
Comment (AlexVlx) [inline tests/CMakeLists.txt:49]: Please match style.

File: tests/CMakeLists.txt [hunk 19]
```diff
@@ -1,82 +1,97 @@
...
+            WIN32_LEAN_AND_MEAN
+            NOMINMAX
+        >
+)
+
+add_library(perf_tests_common INTERFACE)
+
+target_compile_options(perf_tests_common
+    INTERFACE
+        # CMake 3.19 required for list in $<CONFIG:cfg1,cfg2> for shorthand
```
Comment (AlexVlx) [inline tests/CMakeLists.txt:33]: I think even for something that is as daring as HIP-CPU, taking a dependency on CMake 3.19 is a bit too much since various upstream repositories are not quite there yet and asking the user to deal with it would add needless friction, so this should be refactored, even though it's sweet sugar.

File: src/CMakeLists.txt [hunk 20]
```diff
@@ -1 +1,35 @@
+set(CMAKE_CXX_EXTENSIONS OFF)
+
+add_library(src_common INTERFACE)
+
+target_link_libraries(src_common
+    INTERFACE
+        hip_cpu_rt::hip_cpu_rt
+)
+target_compile_options(src_common
```
Comment (AlexVlx) [inline src/CMakeLists.txt:9]: Probably don't need this.

File: src/CMakeLists.txt [hunk 21]
```diff
@@ -1 +1,35 @@
...
+            /arch:AVX2
+            /permissive-
+            /Qvec-report:2
+            /Zc:preprocessor
+        >
+        $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:
+            -Wall
+        >
+)
+target_compile_definitions(src_common
```
Comment (AlexVlx) [inline src/CMakeLists.txt:22]: We do need this, but please match style.

File: src/hip/CMakeLists.txt [hunk 22]
```diff
@@ -1,40 +1,25 @@
-set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_EXTENSIONS OFF)
 
-find_package(Threads REQUIRED)
+add_executable(public_interface_tests
```
Comment (AlexVlx) [inline src/hip/CMakeLists.txt:3]: Please match style.

File: src/hip/CMakeLists.txt [hunk 23]
```diff
@@ -1,40 +1,25 @@
...
-    target_compile_definitions(
-        public_interface_tests PRIVATE NOMINMAX WIN32_LEAN_AND_MEAN)
-endif ()
-target_include_directories(
-    public_interface_tests PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/../../include)
-target_link_libraries(public_interface_tests PRIVATE Threads::Threads)
-
-add_test(NAME public_interface_tests COMMAND public_interface_tests)
\ No newline at end of file
+add_test(
```
Comment (AlexVlx) [inline src/hip/CMakeLists.txt:22]: Please match style.

File: tests/CMakeLists.txt [hunk 24]
```diff
@@ -1,82 +1,97 @@
...
-if (${CMAKE_HOST_APPLE}) # Add brew-ed TBB.
-    include_directories(/usr/local/include)
-    link_directories(/usr/local/lib)
-endif()
-
-add_executable(performance_tests catch_benchmarking_main.cpp benchmarks.cpp)
-if (${CMAKE_CXX_COMPILER_ID} STREQUAL MSVC)
-    target_compile_options(
-        performance_tests PRIVATE
+target_link_libraries(tests_common
```
Comment (AlexVlx) [inline tests/CMakeLists.txt:5]: Please match style.

File: tests/CMakeLists.txt [hunk 25]
```diff
@@ -1,82 +1,97 @@
...
+    INTERFACE
+        hip_cpu_rt::hip_cpu_rt
+)
+target_compile_options(tests_common
```
Comment (AlexVlx) [inline tests/CMakeLists.txt:9]: Probably don't need this.

File: tests/CMakeLists.txt [hunk 26]
```diff
@@ -1,82 +1,97 @@
...
-endif ()
-target_include_directories(
-    performance_tests PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/../include)
-target_link_libraries(performance_tests PRIVATE Threads::Threads)
+        >
+        $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:
+            -Wall
+        >
+)
+target_compile_definitions(tests_common
```
Comment (AlexVlx) [inline tests/CMakeLists.txt:21]: We do need this but please match style.

**Review by AlexVlx — CHANGES_REQUESTED** [review]

**Comment by AlexVlx** [comment]
@MathiasMagnus thank you again for submitting this and apologies for the long delay in reviewing it. In general LGTM, but there are a few things that would increase tidiness, so please go over the comments.

**Comment by MathiasMagnus** [comment]
Thanks, will fix the issues once I get there. rocPRIM will soon land experimental HIP-CPU support, currently with many algos failing, just getting everything to compile using MSVC/GCC/Clang. (Some simple tests do pass). Perhaps at one point HIP-CPU can adopt rocPRIM into it's CI for some massive testing.

**Comment by AlexVlx** [comment]
> Thanks, will fix the issues once I get there. rocPRIM will soon land experimental HIP-CPU support, currently with many algos failing, just getting everything to compile using MSVC/GCC/Clang. (Some simple tests do pass). Perhaps at one point HIP-CPU can adopt rocPRIM into it's CI for some massive testing.

Thank you. Regarding rocPRIM, it'll definitely find its way into the CI infrastructure, the more the merrier:)

**Comment by MathiasMagnus** [comment]
> I think even for something that is as daring as HIP-CPU, taking a dependency on CMake 3.19 is a bit too much since various upstream repositories are not quite there yet and asking the user to deal with it would add needless friction, so this should be refactored, even though it's sweet sugar. 

Indeed, that's just a note that once someone bumps minimum requirements, those lines can be merged.

File: examples/CMakeLists.txt (hunk 1 above)
Comment (MathiasMagnus) [reply examples/CMakeLists.txt]: Hope I got the styling right.

File: examples/CMakeLists.txt (hunk 2 above)
Comment (MathiasMagnus) [reply examples/CMakeLists.txt]: Good.

File: examples/CMakeLists.txt (hunk 3 above)
Comment (MathiasMagnus) [reply examples/CMakeLists.txt]: Done.

File: examples/bit_extract/CMakeLists.txt (hunk 4 above)
Comment (MathiasMagnus) [reply examples/bit_extract/CMakeLists.txt]: Done.

File: examples/hip_bus_bandwidth/CMakeLists.txt (hunk 5 above)
Comment (MathiasMagnus) [reply examples/hip_bus_bandwidth/CMakeLists.txt]: Done.

File: src/hip/CMakeLists.txt (hunk 22 above)
Comment (MathiasMagnus) [reply src/hip/CMakeLists.txt:3]: Done.

File: src/CMakeLists.txt (hunk 21 above)
Comment (MathiasMagnus) [reply src/CMakeLists.txt:22]: Done.

File: src/hip/CMakeLists.txt (hunk 23 above)
Comment (MathiasMagnus) [reply src/hip/CMakeLists.txt:20]: Done.

File: tests/CMakeLists.txt (hunk 24 above)
Comment (MathiasMagnus) [reply tests/CMakeLists.txt]: Done.

File: tests/CMakeLists.txt (hunk 26 above)
Comment (MathiasMagnus) [reply tests/CMakeLists.txt]: Done.

File: examples/hip_dispatch_latency/CMakeLists.txt (hunk 7 above)
Comment (MathiasMagnus) [reply examples/hip_dispatch_latency/CMakeLists.txt]: Done.

File: examples/hip_info/CMakeLists.txt (hunk 9 above)
Comment (MathiasMagnus) [reply examples/hip_info/CMakeLists.txt]: Done.

File: examples/matrix_transpose/CMakeLists.txt (hunk 10 above)
Comment (MathiasMagnus) [reply examples/matrix_transpose/CMakeLists.txt]: Done.

File: examples/pragma_unroll/CMakeLists.txt (hunk 12 above)
Comment (MathiasMagnus) [reply examples/pragma_unroll/CMakeLists.txt]: Done.

File: examples/shared_memory/CMakeLists.txt (hunk 13 above)
Comment (MathiasMagnus) [reply examples/shared_memory/CMakeLists.txt]: Done.

File: examples/square/CMakeLists.txt (hunk 14 above)
Comment (MathiasMagnus) [reply examples/square/CMakeLists.txt]: Done.

File: examples/streams/CMakeLists.txt (hunk 15 above)
Comment (MathiasMagnus) [reply examples/streams/CMakeLists.txt]: Done.

File: examples/vadd_hip/CMakeLists.txt (hunk 16 above)
Comment (MathiasMagnus) [reply examples/vadd_hip/CMakeLists.txt:5]: Done.

File: tests/CMakeLists.txt (hunk 17 above)
Comment (MathiasMagnus) [reply tests/CMakeLists.txt]: Done.

File: tests/CMakeLists.txt (hunk 18 above)
Comment (MathiasMagnus) [reply tests/CMakeLists.txt]: Done.

File: tests/CMakeLists.txt (hunk 19 above)
Comment (MathiasMagnus) [reply tests/CMakeLists.txt:17]: Indeed, that's just a note that once someone bumps minimum requirements, those lines can be merged.

File: src/CMakeLists.txt (hunk 20 above)
Comment (MathiasMagnus) [reply src/CMakeLists.txt:9]: Done.

File: tests/CMakeLists.txt (hunk 25 above)
Comment (MathiasMagnus) [reply tests/CMakeLists.txt]: Indeed, that's just a note that once someone bumps minimum requirements, those lines can be merged.

**Review by MathiasMagnus — COMMENTED** [review]
I think I addressed all issues, plus rebased my work on to tip of master.

**Comment by AlexVlx** [comment]
@MathiasMagnus thank you very much for this and for making the changes (I did a wee bit of tinkering out of pure pedantry, I hope you don't mind), but overall LGTM!

**Comment by MathiasMagnus** [comment]
No problem. I couldn't quite figure out the style you were looking for based on the textual desc and the root CMakelists.txt. Maybe one day we'll have semantic aware "Clang-tidy" for CMake scripts.
